lib/ansible/modules/cloud/amazon/ec2_transit_gateway_info.py

```diff
@@ -0,0 +1,263 @@
+                 dns_support:
+                    description:
+                      - Indicates whether DNS support is enabled.
+                    returned: always
+                    type: str
+                    sample: "enable"
+                 propagation_default_route_table_id:
+                    description:
+                      - The ID of the default propagation route table.
+                    returned: when present
+                    type: str
+                    sample: "rtb-11223344"
+                 vpn_ecmp_support:
+                    description:
+                      - Indicates  whether  Equal Cost Multipath Protocol support
+                        is enabled.
+                    returned: always
+                    type: str
+                    sample: "enable"
+        owner_id:
+            description: The AWS account number ID which owns the transit gateway.
+            returned: always
+            type: str
+            sample: "1234567654323"
+        state:
+            description: The state of the transit gateway.
+            returned: always
+            type: str
+            sample: "available"
+        tags:
+            description: A dict of tags associated with the transit gateway.
+            returned: always
+            type: dict
+            sample:
+              Name: "A sample TGW"
+        transit_gateway_arn:
+            description: The Amazon Resource Name (ARN) of the transit gateway.
+            returned: always
+            type: str
+            sample: "arn:aws:ec2:us-west-2:1234567654323:transit-gateway/tgw-02c42332e6b7da829"
+        transit_gateway_id:
+            description: The ID of the transit gateway.
+            returned: always
+            type: str
+            sample: "tgw-02c42332e6b7da829"
+'''
+
+try:
+    from botocore.exceptions import BotoCoreError, ClientError
+except Exception:
+    pass
+    # handled by imported AnsibleAWSModule
+
+from ansible.module_utils.aws.core import AnsibleAWSModule
+from ansible.module_utils.ec2 import (
+    AWSRetry,
+    boto3_tag_list_to_ansible_dict,
+    camel_dict_to_snake_dict,
+    ansible_dict_to_boto3_filter_list
+)
+
+
+class AnsibleEc2TgwInfo(object):
+
+    def __init__(self, module, results):
+        self._module = module
+        self._results = results
+        self._connection = self._module.client('ec2')
+        self._check_mode = self._module.check_mode
+
+        if not hasattr(self._connection, 'describe_transit_gateways'):
+            self._module.fail_json(msg='transit gateway module requires boto3 >= 1.9.52')
+
+    @AWSRetry.exponential_backoff()
+    def describe_transit_gateways(self):
+        """
+        Describe transit gateways.
+
+        module  : AnsibleAWSModule object
+        connection  : boto3 client connection object
+        """
+        # collect parameters
+        filters = ansible_dict_to_boto3_filter_list(self._module.params.get('filters'))
```

Comment:
```suggestion
        filters = ansible_dict_to_boto3_filter_list(self._module.params['filters'])
```
(if you don't prefer `get()` for whatever reason)

Reply:
guess overkill since there is a default {} already.

Reply:
No, more overkill because everything specified in argspec will be contained in `module.params`. If it has no default and is not specified by the user, it will have value `None`.

Reply:
.get() removed